virtual_ecosystem/models/soil/pools.py

```diff
@@ -512,21 +514,11 @@ def calculate_all_pool_updates(
             secondary_p_breakdown_rate=self.constants.secondary_phosphorus_breakdown_rate,
             labile_p_sorption_rate=self.constants.labile_phosphorus_sorption_rate,
         )
-        # Convert ammonium and phosphorus depositions from per area to per volume units,
-        # under the assumption the phosphorus just gets deposited in the very upper
-        # layer of the soil
-        ammonium_deposition = (
-            self.constants.ammonium_deposition_rate
-            / self.max_depth_of_microbial_activity
-        )
-        phosphorus_deposition = (
-            self.constants.phosphorus_deposition_rate
-            / self.max_depth_of_microbial_activity
-        )
 
         # Determine net changes to the pools
```

Comment:
So this is where each component in a summuation can lead to negative delta values?

Reply:
Yep it's in this section. So e.g. `self.data["plant_ammonium_uptake"]` does not change as the soil model progresses (as it's calculated by the plants model), so it will still be subtracted at the same rate even "soil_n_pool_ammonium" has decreased due to soil model processes, which means that it can potentially drive the value negative